swagger.yaml

```diff
@@ -532,6 +532,119 @@ paths:
         "default":
           description: Software entry not found
 
+  "/questionnaire":
+    get:
+      operationId: listQuestionnaires
+      summary: Get all questionnaire entries
+      description: Retrieve a list of all questionnaire entries in the registry.
+      tags:
+        - Questionnaire
+      responses:
+        "200":
+          description: A list of questionnaire entries
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Questionnaire"
+
+    post:
+      operationId: createQuestionnaire
+      summary: Register new questionnaire
+      description: Register a new questionnaire entry in the registry.
+      tags:
+        - Questionnaire
+      requestBody:
+        description: Questionnaire entry to register
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Questionnaire"
+      responses:
+        "201":
+          description: Questionnaire entry created successfully
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Questionnaire"
+        "default":
+          description: Bad request, questionnaire entry already exists or invalid data
```

Comment:
I think this "default" response and all the other endpoints (including the list which does not currently have the default response) could be improved by using something like this:
"default":
          description: Invalid request was sent to the server.
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: "#/components/schemas/Error"
                  
It's taken from the time-bank-v2-spec and would require adding an error schema